src/components/Modal/index.js

```diff
@@ -1,110 +1,213 @@
 import React from "react";
 import PropTypes from "prop-types";
-import styled from "styled-components";
-
-import CrossIcon from "../Icons/Cross";
-import IconButton from "../Button/IconButton";
-import { spacing, colors, constants } from "../../theme/index";
-import { ItemContainerConsumer } from "../List/Context";
-import { BackdropConsumer } from "../Backdrop/Context";
-import Column from "../Grid/Column";
-import Row from "../Grid/Row";
-
-const ModalContainer = styled(Column)`
-  background-color: ${colors.white.base};
-  border-radius: ${constants.borderRadius.large};
-  box-shadow: 0 16px 16px 0 rgba(0, 0, 0, 0.06), 0 0 16px 0 rgba(0, 0, 0, 0.12);
-  border: solid 1px rgba(0, 0, 0, 0.04);
-  padding: 0;
-  overflow-y: auto;
-  overflow-x: hidden;
-  max-height: calc(100% - 96px);
-  z-index: 100;
-
-  &.modal-enter {
-    opacity: 0;
-    transform: scale(0.7);
-  }
 
-  &.modal-enter.modal-enter-active {
-    opacity: 1;
-    transform: scale(1);
-    transition: all 0.3s ease-in-out;
-  }
+import {
+  ModalContainer,
+  ModalContent,
+  ActionBar,
+  BottomActionBar
+} from "./index.styles";
 
-  &.modal-leave {
-    opacity: 1;
-    transform: scale(1);
-  }
+import Backdrop from "../Backdrop";
+import DefaultActionBar from "./DefaultActionBar";
+
+import { ModalProvider } from "./context";
+import {
+  MODAL_SIZE_SMALL,
+  MODAL_SIZE_MEDIUM,
+  MODAL_SIZE_LARGE,
+  getModalSize
+} from "./size";
+import { withDeviceSize } from "../DeviceSize/Context";
+import {
+  getContentHeight,
+  getBottomActionBarShadow,
+  getActionBarShadow,
+  isRequestCloseApproved
+} from "./helper";
 
-  &.modal-leave.modal-leave-active {
-    opacity: 0;
-    transition: all 0.1s ease-in-out;
-    transform: scale(0.7);
+class Modal extends React.Component {
```

Comment:
Do these changes have any implications for the `ListContainer` use case of the `Modal`?

Reply:
Nope. This modal doesn't break anything related to the ListContainer's implementation of Modal (obsolete implementation was moved to the List component directory)